WordPress/src/main/java/org/wordpress/android/util/image/getters/WPRemoteResourceViewTarget.kt

```diff
@@ -0,0 +1,135 @@
+package org.wordpress.android.util.image.getters
+
+import android.annotation.SuppressLint
+import android.graphics.Canvas
+import android.graphics.ColorFilter
+import android.graphics.PixelFormat
+import android.graphics.Rect
+import android.graphics.drawable.Animatable
+import android.graphics.drawable.Drawable
+import android.widget.TextView
+
+import com.bumptech.glide.request.Request
+import com.bumptech.glide.request.target.SizeReadyCallback
+import com.bumptech.glide.request.target.Target
+import com.bumptech.glide.request.target.ViewTarget
+import com.bumptech.glide.request.transition.Transition
+
+import org.wordpress.android.util.R
+
+/**
+ * A class that we can load a remote resource into. Automatically displays placeholder while the remote img is
+ * loading and displays an error image if the loading fails.
+ *
+ * We could probably subclass BaseTarget instead of ViewTarget, since we basically override most of its functionality.
+ * However, we might want to use ViewTarget.clearOnDetach(..) when it becomes stable (it's experimental now).
+ * It clears the View's Request when the View is detached from its Window and restarts the Request when the View is
+ * re-attached from its Window.
+ */
+internal class WPRemoteResourceViewTarget(
+    view: TextView,
+    private val maxSize: Int
+) : ViewTarget<TextView, Drawable>(view) {
+    private val drawableWrapper = RemoteDrawableWrapper()
+    private var request: Request? = null
+
+    val drawable: Drawable get() = drawableWrapper
+
+    override fun onResourceReady(resource: Drawable, transition: Transition<in Drawable>?) {
+        if (resource is Animatable) {
+            // Bind a Callback object to this Drawable.  Required for clients that want to support
+            // animated drawables.
+            resource.callback = getView().getTag(R.id.glide_image_loader_view_tag) as Drawable.Callback
+            (resource as Animatable).start()
+        }
+        replaceDrawable(resource, getScaledBounds(resource, maxSize))
+    }
+
+    override fun onLoadFailed(errorDrawable: Drawable?) {
+        errorDrawable?.let {
+            replaceDrawable(it, Rect(0, 0, it.intrinsicWidth, it.intrinsicHeight))
+        }
+    }
+
+    override fun onLoadStarted(res: Drawable?) {
+        super.onLoadStarted(res)
+        res?.let {
+            replaceDrawable(it, Rect(0, 0, it.intrinsicWidth, it.intrinsicHeight))
+        }
+    }
+
+    private fun getScaledBounds(resource: Drawable, maxWidth: Int): Rect {
+        val imgWidth = resource.intrinsicWidth
+        val imgHeight = resource.intrinsicHeight
+        val xScale = imgWidth.toFloat() / maxWidth.toFloat()
```

Comment:
To simplify the calculation below, we can simply do `Math.max(1.0f, imgWidth.toFloat() / maxWidth.toFloat())`.

Reply:
Good call;). I've just copied the code from WPImageGetter. I should have refactored it a bit.